dap-etl/extraction/src/main/scala/org/broadinstitute/monster/dap/environment/EnvironmentExtractionPipeline.scala

```diff
@@ -47,7 +49,16 @@ object EnvironmentExtractionPipeline extends ScioApp[Args] {
     if (start.isAfter(end)) throw new EnvironmentExtractionFailException
     val dateList =
       (Iterator.iterate(start)(_ plusDays 1) takeWhile (_ isBefore end.plusDays(1))).toList
-    dateList.map(date => date.format(formatter).toLowerCase).distinct
+    dateList
+      .map(date =>
+        // event names for June, July, and Sept have 4 alpha characters before the year
+        if (List(6, 7, 9).contains(date.getMonthValue)) {
```

Comment:
What I wrote feels very hacky - please feel free to suggest and revisions of this logic

Reply:
Given the non-standard abbreviation used by the surveys, I think the alternative here is to use a `DateTimeFormatterBuilder`,  see [this](https://stackoverflow.com/questions/30816016/how-to-parse-non-standard-month-names-with-datetimeformatter) stackoverflow answer.

Something like this:
```
  val monthMap: java.util.Map[java.lang.Long, String] = Map(
      Long.box(4L) -> "Apr",
      Long.box(6L) -> "June",
      Long.box(7L) -> "July",
      Long.box(9L) -> "Sept"
// ... we'll need all 12 months here
    ).asJava
    val formatter = new DateTimeFormatterBuilder()
      .appendText(ChronoField.MONTH_OF_YEAR, monthMap)
      .appendPattern("yyyy")
      .toFormatter()

    val april = OffsetDateTime.parse("2021-04-03T10:15:30+01:00")
    val june = OffsetDateTime.parse("2021-06-03T10:15:30+01:00")
    val july = OffsetDateTime.parse("2021-07-03T10:15:30+01:00")
    val sept = OffsetDateTime.parse("2021-09-03T10:15:30+01:00")
```

This may be marginally cleaner at the expense of our own custom months map. Worth a try.

Reply:
Implemented a DateTimeFormatterBuilder based on the provided example.
The new logic passed the existing unit tests:
`testOnly org.broadinstitute.monster.dap.environment.EnvironmentExtractionPipelineSpec`